### PR TITLE
Add basic layer modes description, including transform button

### PR DESCRIPTION
**Pull request by dalthviz**
# References and relevant issues

Depends on https://github.com/napari/napari/pull/6794

# Description

Adds new content related with layers transform mode and base possible values (`pan_zoom` and `transform`)

### CODE REVIEW

File: docs/guides/layers.md [hunk 1]
```diff
@@ -28,6 +28,32 @@ currently supported by napari, check out the
 introduction, check out the
 [Layer list section in the napari viewer tutorial](layer-list).
 
+(layer_mode)=
+## Layer mode
+
+All our layers support a `mode` property that changes the way you interact
+with the layer from the viewer. These modes are accesible via the layer controls
```
Comment (melissawm) [inline docs/guides/layers.md]:
```suggestion
with the layer from the viewer. These modes are accessible via the layer controls
```

File: docs/guides/layers.md [hunk 2]
```diff
@@ -28,6 +28,32 @@ currently supported by napari, check out the
...
+widget top buttons and via shortcuts (pressing a number from `1` to the number of
+modes available for the layer). Depending on the layer type, more modes could be
+available. Currently, there are two base modes:
```
Comment (melissawm) [inline docs/guides/layers.md]: I think this is confusing - the sentence makes it seem like for any layer type, pressing 1 would activate the pan/zoom or transform modes and that's not the case. Maybe we don't need to mention a number here and just say

```suggestion
widget top buttons and via keyboard shortcuts (these shortcuts will vary depending on the layer type). Currently, there are two base modes:
```

File: docs/howtos/layers/shapes.md [hunk 3]
```diff
@@ -171,13 +172,20 @@ are used. i.e. You can't remove a vertex before you have created a shape.
...
   is disabled when using the adding and editing tools. Temporarily re-enable pan
   and zoom by pressing and holding the spacebar. This feature can be useful if
   you want to move around the shapes layer as you edit it.
 
+* **Transform**
+  ![image: Transform](../../images/transform-tool.png)
+
+  Use this tool to rotate, scale, or translate the layer. To reset the transformation,
+  you can Option/Alt-click the transform button (a confirmation dialog will open to
+  confirm the reset).
```
Comment (melissawm) [inline docs/howtos/layers/shapes.md]: The `6` and `7` keyboard shortcuts are also valid here.

**Review by melissawm — COMMENTED** [review]
Did an initial pass, only a couple comments. Thanks @dalthviz !

File: docs/howtos/layers/shapes.md (hunk 3 above)
Comment (dalthviz) [reply docs/howtos/layers/shapes.md]: Hi @melissawm thanks for the review! You mean changing this to be something like:

```
 Use this tool to rotate, scale, or translate the layer. To activate the tool, you can
 press `7`. To reset the transformation, you can Option/Alt-click the transform
 button (a confirmation dialog will open to confirm the reset).
```

?

Comment (melissawm) [reply docs/howtos/layers/shapes.md]: Yes - and add the information about the keyboard shortcut to the pan/zoom tool as well if you can. Thanks!

**Review by melissawm — APPROVED** [review]
Looks good to me, thanks @dalthviz !

File: docs/guides/layers.md [hunk 4]
```diff
@@ -28,6 +28,30 @@ currently supported by napari, check out the
...
+
+All our layers support a `mode` property that changes the way you interact
+with the layer from the viewer. These modes are accessible via the layer controls
+widget top buttons and via keyboard shortcuts (these shortcuts will vary depending on the layer type). Currently, there are two base modes:
+
+* Pan and zoom
+![image: Pan/zoom](../images/pan-zoom-tool.png)
+
+The `pan_zoom` mode allows you to pan around and zoom in/out the layer. It's
+the default mode selected.
```
Comment (psobolewskiPhD) [inline docs/guides/layers.md]:
```suggestion
the default mode selected. Note: zooming with the mouse-wheel should essentially always work!
```

File: docs/guides/layers.md [hunk 5]
```diff
@@ -28,6 +28,30 @@ currently supported by napari, check out the
...
 
+(layer_mode)=
+## Layer mode
+
+All our layers support a `mode` property that changes the way you interact
+with the layer from the viewer. These modes are accessible via the layer controls
+widget top buttons and via keyboard shortcuts (these shortcuts will vary depending on the layer type). Currently, there are two base modes:
+
+* Pan and zoom
+![image: Pan/zoom](../images/pan-zoom-tool.png)
```
Comment (psobolewskiPhD) [inline docs/guides/layers.md:39]: Could we use the icon that's in 0.4.19, the pan icon?
https://github.com/napari/napari/blob/main/napari/resources/icons/pan_arrows.svg
It's confusing that the mode is called `pan_zoom`, but zoom essentially always works with the mouse wheel, so it's really `pan` that's unique to the mode.

File: docs/guides/layers.md [hunk 6]
```diff
@@ -28,6 +28,30 @@ currently supported by napari, check out the
...
+
+The `pan_zoom` mode allows you to pan around and zoom in/out the layer. It's
+the default mode selected.
+
+* Transform
+![image: Transform](../images/transform-tool.png)
+
+The `transform` mode allows you to translate, rotate, and scale the layer
+graphically. To reset the transformation, you can Option/Alt-click the transform
```
Comment (psobolewskiPhD) [inline docs/guides/layers.md]:
```suggestion
graphically. Note: While you can transform 3D layers, at present this mode is only usable in 2D viewer display mode.
To reset the transformation, you can Option/Alt-click the transform
```

File: docs/howtos/layers/image.md [hunk 7]
```diff
@@ -25,19 +25,37 @@ For more information about layers, refer to [Layers at a glance](../../guides/la
...
 sample image of your choice.
 
+### Buttons
+
+* `Pan/zoom` ![image: Pan/zoom tool](../../images/pan-zoom-tool.png) is the default mode
+  of the layer and supports panning and zooming. Press the `1` key when the layer is selected
+  to use this mode.
+
+* `Transform` ![image: Transform](../../images/transform-tool.png) enables you to
+  rotate, scale, or translate the layer. To reset the transformation,
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/image.md]:
```suggestion
  rotate, scale, or translate the layer. Note: at present this feature is limited to 2D viewer display mode.
  To reset the transformation,
```

File: docs/howtos/layers/labels.md [hunk 8]
```diff
@@ -256,6 +257,14 @@ pan and zoom is selected, editing the layer is not possible. Once you click on
 one of the editing tools, pan and zoom is turned off. Return to pan and zoom
 mode by pressing the `6` key when the `labels layer` is selected.
 
+### Transform mode
+
+This mode is represented by ![image: Transform](../../images/transform-tool.png) in the
+`layer controls` panel. It enables you to rotate, scale, or translate the layer.
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/labels.md:263]:
```suggestion
`layer controls` panel. It enables you to rotate, scale, or translate the layer.
Note: at present this feature is limited to 2D viewer display mode.
```

File: docs/howtos/layers/points.md [hunk 9]
```diff
@@ -125,6 +126,14 @@ layer:
   options are supported only when viewing a layer using 2D rendering. Return to
   pan and zoom mode by pressing the `4` key when the points layer is selected.
 
+* **Transform**
+  ![image: Transform](../../images/transform-tool.png)
+
+  Use this tool to rotate, scale, or translate the layer. To reset the transformation,
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/points.md]:
```suggestion
  Use this tool to rotate, scale, or translate the layer. 
  Note: at present this feature is limited to 2D viewer display mode. To reset the transformation,
```

File: docs/howtos/layers/shapes.md [hunk 10]
```diff
@@ -171,12 +172,20 @@ are used. i.e. You can't remove a vertex before you have created a shape.
...
+  Use this tool to pan around the layer or zoom in. Pan and zoom functionality
+  is disabled when using the adding and editing tools. To activate the tool,
+  you can press `6`. Temporarily re-enable pan and zoom by pressing and holding
+  the spacebar. This feature can be useful if you want to move around the shapes
+  layer as you edit it.
+
+* **Transform**
+  ![image: Transform](../../images/transform-tool.png)
+
+  Use this tool to rotate, scale, or translate the layer. To activate the tool,
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/shapes.md]:
```suggestion
  Use this tool to rotate, scale, or translate the layer. Note: at present this feature is limited to 2D viewer display mode. To activate the tool,
```

File: docs/howtos/layers/shapes.md [hunk 11]
```diff
@@ -171,12 +172,20 @@ are used. i.e. You can't remove a vertex before you have created a shape.
...
 * **Pan/zoom**
-  ![image: Select shapes](../../images/pan-zoom-tool.png)
+  ![image: Pan/zoom](../../images/pan-zoom-tool.png)
 
-  Use this tool to pan around the image or zoom in. Pan and zoom functionality
-  is disabled when using the adding and editing tools. Temporarily re-enable pan
-  and zoom by pressing and holding the spacebar. This feature can be useful if
-  you want to move around the shapes layer as you edit it.
+  Use this tool to pan around the layer or zoom in. Pan and zoom functionality
+  is disabled when using the adding and editing tools. To activate the tool,
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/shapes.md]:
```suggestion
  Use this tool to pan around the layer or zoom in. Pan functionality
  is disabled when using the adding and editing tools; however, zoom using the mouse-wheel will typically continue to work. To activate the tool,
```

File: docs/howtos/layers/surface.md [hunk 12]
```diff
@@ -77,20 +77,29 @@ viewer.close()
...
-* Blending - Choose `opaque`, `translucent`, `translucent no depth`, or
-  `additive` from the dropdown. Refer to the [Blending layers](blending-layers)
-  section of _Layers at a glance_ for an explanation of each type of blending.
-* Shading - Choose `none`, `flat`, or `smooth` from the dropdown.
+* **Buttons**
+  * Pan/zoom - ![image: Pan/zoom tool](../../images/pan-zoom-tool.png) is the default
+    mode of the layer and supports panning and zooming. Press the `1` key when the
+    layer is selected to use this mode.
+  * Transform - ![image: Transform](../../images/transform-tool.png) enables you to
+    rotate, scale, or translate the layer. To reset the transformation, you can
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/surface.md]:
```suggestion
    rotate, scale, or translate the layer. Note: at present this feature is limited to 2D viewer display mode. To reset the transformation, you can
```

File: docs/howtos/layers/tracks.md [hunk 13]
```diff
@@ -92,34 +92,43 @@ napari.run()
...
-  each track. Assigning values to `track_id` is explained in
-  [Tracks data](#tracks-data) below.
-* Graph - check this box to display a previously created graph as explained in
-  [](#arguments-of-view_tracks-and-add_tracks).
+* **Buttons**
+  * Pan/zoom - ![image: Pan/zoom tool](../../images/pan-zoom-tool.png) is the default
+    mode of the layer and supports panning and zooming. Press the `1` key when the
+    layer is selected to use this mode.
+  * Transform - ![image: Transform](../../images/transform-tool.png) enables you to
+    rotate, scale, or translate the layer. To reset the transformation, you can
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/tracks.md]:
```suggestion
    rotate, scale, or translate the layer. Note: at present this feature is limited to 2D viewer display mode. To reset the transformation, you can
```

File: docs/howtos/layers/vectors.md [hunk 14]
```diff
@@ -85,29 +85,38 @@ Before you can use the GUI to manipulate vectors, you must load a vector layer.
...
-  color from the pallette.
-* Out of slice - if this box is checked, `out of slice` is on or true. If this
-  box is not checked, `out of slice` is off or false. If it is on or true,
-  vectors slightly out of slice are rendered.
+* **Buttons**
+  * Pan/zoom - ![image: Pan/zoom tool](../../images/pan-zoom-tool.png) is the default
+    mode of the layer and supports panning and zooming. Press the `1` key when the
+    layer is selected to use this mode.
+  * Transform - ![image: Transform](../../images/transform-tool.png) enables you to
+    rotate, scale, or translate the layer. To reset the transformation, you can
```
Comment (psobolewskiPhD) [inline docs/howtos/layers/vectors.md]:
```suggestion
    rotate, scale, or translate the layer. Note: at present this feature is limited to 2D viewer display mode. To reset the transformation, you can
```

**Review by psobolewskiPhD — CHANGES_REQUESTED** [review]
Thanks @dalthviz 
I made a few suggestions to clarify that translate only works in 2D viewer mode.
I'm still not sure we should be even more explicit that it *works* on 3D **layers** but only in 2D viewer mode?
Also I tried to clarify the pan/zoom bit -- I'd prefer we use the 0.4.19 arrows/pan icon.

**Comment by dalthviz** [comment]
> I'm still not sure we should be even more explicit that it works on 3D layers but only in 2D viewer mode?

I would say the added note should be enough but maybe asking for feedback to others could be worthy

**Review by psobolewskiPhD — APPROVED** [review]
Thanks @dalthviz !
Looks good to me.

**Pull request merged**